### PR TITLE
KAA-1300: Administration Guide -> System configuration -> General configuration

### PR DESCRIPTION
**Pull request by sashadidukh**

### CODE REVIEW

File: doc/Administration-guide/System-Configuration/General-configuration/index.md [hunk 1]
```diff
@@ -4,4 +4,183 @@ title: General configuration
 permalink: /:path/
 sort_idx: 10
 ---
-Kaa node configuration. Inter-node communication and networking config.
+
+After Kaa installation on Ubuntu/Debian OS (deb packages), configuration files for each Kaa component will be extracted into the
```
Comment (ashvayka) [inline doc/Administration-guide/System-Configuration/General-configuration/index.md]: Please check installation guide and add (if this is different) info for CentOS/RHEL

File: doc/Administration-guide/System-Configuration/General-configuration/index.md [hunk 2]
```diff
@@ -4,4 +4,183 @@ title: General configuration
...
+
+#### operations_service_enabled
+Default: _true_
+
+Specifies if Operations Service is enabled.
+
+#### thrift_host
+Default: _localhost_
+
+Thrift service host address.
```
Comment (ashvayka) [inline doc/Administration-guide/System-Configuration/General-configuration/index.md]: This information is used for Thrift remote procedure calls between nodes in Kaa cluster.

File: doc/Administration-guide/System-Configuration/General-configuration/index.md [hunk 3]
```diff
@@ -4,4 +4,183 @@ title: General configuration
...
+
+#### thrift_port
+Default: _9090_
+
+Thrift service port address.
```
Comment (ashvayka) [inline doc/Administration-guide/System-Configuration/General-configuration/index.md]: This information is used for Thrift remote procedure calls between nodes in Kaa cluster.

File: doc/Administration-guide/System-Configuration/General-configuration/index.md [hunk 4]
```diff
@@ -4,4 +4,183 @@ title: General configuration
...
+
+#### admin_port
+Default: _8080_
+
+Kaa Admin Web service port.
```
Comment (ashvayka) [inline doc/Administration-guide/System-Configuration/General-configuration/index.md]: Kaa Administation Web UI port.

File: doc/Administration-guide/System-Configuration/General-configuration/index.md [hunk 5]
```diff
@@ -4,4 +4,183 @@ title: General configuration
...
+
+#### zk_enabled
+Default: _true_
+
+Specifies if need to use zookeeper service. This is property have to be always _true_. It is possible to change it for development or debug process.
+
+#### zk_host_port_list
+Default: _localhost:2181_
+
+Comma-separated url list of Zookeeper nodes: hostname:port.
```
Comment (ashvayka) [inline doc/Administration-guide/System-Configuration/General-configuration/index.md]: Comma-separated url list of Zookeeper nodes: hostname1:port1,hostname2:port2

File: doc/Administration-guide/System-Configuration/General-configuration/index.md [hunk 6]
```diff
@@ -4,4 +4,183 @@ title: General configuration
...
+
+#### zk_max_retry_time
+Default: _3000_
+
+The max retry time in milliseconds.
```
Comment (ashvayka) [inline doc/Administration-guide/System-Configuration/General-configuration/index.md]: retry of what operations? Add this to the doc

File: doc/Administration-guide/System-Configuration/General-configuration/index.md [hunk 7]
```diff
@@ -4,4 +4,183 @@ title: General configuration
...
+#### loadmgmt_max_session_redirect_probability
+Default: _0.0_
+
+Maximum redirect probability for existing sessions.
+
+
+#### recalculation_period
+Default: _10_
+
+Recalculate period in seconds for Operations service load process.
```
Comment (ashvayka) [inline doc/Administration-guide/System-Configuration/General-configuration/index.md]: Recalculate period in seconds for Operations service load balancer process

File: doc/Administration-guide/System-Configuration/General-configuration/index.md [hunk 8]
```diff
@@ -4,4 +4,183 @@ title: General configuration
...
+
+#### user_hash_partitions
+Default: _10_
+
+Specify hash partitions count for each server node.
```
Comment (ashvayka) [inline doc/Administration-guide/System-Configuration/General-configuration/index.md]: Specify consistent-hash partitions count for each server node.

File: doc/Administration-guide/System-Configuration/General-configuration/index.md [hunk 9]
```diff
@@ -4,4 +4,183 @@ title: General configuration
...
+
+#### ops_server_history_ttl
+Default: _3600_
+
+Default time to live in seconds for historical information about Operations service load.
+
+#### worker_thread_pool
+Default: _8_
+
+Handler thread pool executor size.
```
Comment (ashvayka) [inline doc/Administration-guide/System-Configuration/General-configuration/index.md]: handler of what?

**Comment by sashadidukh** [comment]
Fix comments above

**Comment by ashvayka** [comment]
LGTM except one item. Please check if this property "tmp_keys" is used. If not - please remove it from all main, test and doc resources. Also fix merge conflict.

**Comment by sashadidukh** [comment]
Property "tmp_keys" from kaa-node.properties is not documented. It will be removed from all main, test and doc resources according to [KAA-1343](http://jira.kaaproject.org/browse/KAA-1343).

**Comment by ashvayka** [comment]
LGTM

File: doc/Administration-guide/System-Configuration/General-configuration/index.md [hunk 10]
```diff
@@ -4,4 +4,183 @@ title: General configuration
 permalink: /:path/
 sort_idx: 10
 ---
-Kaa node configuration. Inter-node communication and networking config.
+
+After [Kaa installation]({{root_url}}Administration-guide/System-installation/), configuration files for each Kaa component will be extracted into the
```
Comment (Polinaa) [inline doc/Administration-guide/System-Configuration/General-configuration/index.md:8]: _Page not found_ for all links. Include _variables.md_ file to capture _root_url_.

File: doc/Administration-guide/System-Configuration/General-configuration/index.md [hunk 11]
```diff
@@ -4,4 +4,183 @@ title: General configuration
...
+Default: _localhost_
+
+Thrift service host address. This information is used for Thrift remote procedure calls between nodes in
+[Kaa cluster]({{root_url}}Administration-guide/System-installation/Cluster-setup/).
+
+
+#### thrift_port
+Default: _9090_
+
+Thrift service port address. Thrift service host address. This information is used for Thrift remote procedure calls between nodes in
```
Comment (Polinaa) [inline doc/Administration-guide/System-Configuration/General-configuration/index.md:47]: Remove: _Thrift service host address_.

File: doc/Administration-guide/System-Configuration/General-configuration/index.md [hunk 12]
```diff
@@ -4,4 +4,183 @@ title: General configuration
...
+
+#### transport_public_interface
+Default: _localhost_
+
+Interface that will be reported by all transports.
+
+#### metrics_enabled
+Default: _true_
+
+Specify if metrics collections is enabled. See
```
Comment (Polinaa) [inline doc/Administration-guide/System-Configuration/General-configuration/index.md:158]: are enabled

File: doc/Administration-guide/System-Configuration/General-configuration/index.md [hunk 13]
```diff
@@ -4,4 +4,183 @@ title: General configuration
...
+
+#### admin_port
+Default: _8080_
+
+Kaa Administration Web UI port.
+
+#### zk_enabled
+Default: _true_
+
+Specifies if need to use zookeeper service. This is property have to be always _true_. It is possible to change it for development or debug process.
```
Comment (Polinaa) [inline doc/Administration-guide/System-Configuration/General-configuration/index.md:58]: 1) **Z**ookeeper
2) This property has to be

File: doc/Administration-guide/System-Configuration/General-configuration/index.md [hunk 14]
```diff
@@ -4,4 +4,183 @@ title: General configuration
...
+```bash
+$ sudo service kaa-node restart
+```
+
+The kaa-node.properties consist of the following parameters:
+
+#### control_service_enabled
+Default: _true_
+
+Specifies if Control Service is enabled.
```
Comment (Polinaa) [inline doc/Administration-guide/System-Configuration/General-configuration/index.md:25]: Add references to Control, Bootstrap, Operations, and Zookeeper services.

**Pull request merged**